src/gui/Window.qml

```diff
@@ -22,6 +24,7 @@ ApplicationWindow {
     property var uiState: ({})
     property var history: ({})
     property var theme: null
+    property string configDir
```

Comment:
Avoid using abbreviations in variable/functions/etc names (Qt coding conventions), `settingsFolder` would be better (applies anyway you've written "configDir")